worlds/sm64ex/Rules.py

```diff
@@ -1,33 +1,54 @@
-from ..generic.Rules import add_rule
-from .Regions import connect_regions, sm64_level_to_paintings, sm64_paintings_to_level, sm64_level_to_secrets, sm64_entrances_to_level, sm64_level_to_entrances
+from typing import Callable, Union
 
-def shuffle_dict_keys(world, obj: dict) -> dict:
-    keys = list(obj.keys())
-    values = list(obj.values())
+from BaseClasses import MultiWorld
+from ..generic.Rules import add_rule, set_rule
+from .Locations import location_table
+from .Regions import connect_regions, sm64_level_to_paintings, sm64_paintings_to_level, sm64_level_to_secrets, \
+sm64_level_to_secrets, sm64_entrances_to_level, sm64_level_to_entrances
+from .Items import action_item_table
+
+def shuffle_dict_keys(world, dictionary: dict) -> dict:
+    keys = list(dictionary.keys())
+    values = list(dictionary.values())
     world.random.shuffle(keys)
-    return dict(zip(keys,values))
+    return dict(zip(keys, values))
 
-def fix_reg(entrance_ids, entrance, destination, swapdict, world):
-    if entrance_ids[entrance] == destination: # Unlucky :C
+def fix_reg(entrance_map, entrance, destination, swapdict, world):
+    if entrance_map[swapdict[entrance]] == destination: # Unlucky :C
         rand = world.random.choice(swapdict.keys())
-        entrance_ids[entrance], entrance_ids[swapdict[rand]] = rand, entrance_ids[entrance]
-        swapdict[rand] = entrance_ids[entrance]
+        entrance_map[entrance], entrance_map[swapdict[rand]] = rand, entrance_map[entrance]
+        swapdict[rand] = entrance_map[entrance]
         swapdict.pop(entrance)
 
-def set_rules(world, player: int, area_connections: dict):
+def set_rules(world, player: int, area_connections, star_costs, move_rando_bitvec: int):
```

Comment:
Why did you remove the typing annotation?

Reply:
Accident.  (https://github.com/ArchipelagoMW/Archipelago/pull/2569/commits/007aa8133f9d39072b77531a4f8b9a3749f2147b)